src/codec_aom.c

```diff
@@ -1142,7 +1142,8 @@ static avifResult aomCodecEncodeImage(avifCodec * codec,
             break;
         }
         if (pkt->kind == AOM_CODEC_CX_FRAME_PKT) {
-            avifCodecEncodeOutputAddSample(output, pkt->data.frame.buf, pkt->data.frame.sz, (pkt->data.frame.flags & AOM_FRAME_IS_KEY));
+            AVIF_CHECKRES(
+                avifCodecEncodeOutputAddSample(output, pkt->data.frame.buf, pkt->data.frame.sz, (pkt->data.frame.flags & AOM_FRAME_IS_KEY)));
```

Comment:
This was intended. The `avifResult` is propagated up to `avifEncoderAddImage()` and `AVIF_RESULT_OUT_OF_MEMORY` is both rare and precise enough. `avifDiagnosticsPrintf()` will not add much more details besides the function where it happened anyway.

At line 1191 the return type is boolean, which is why the diagnostics are filled; otherwise there is no way for the user to know that it was a memory allocation failure.